src/zzz_od/operation/battle/shiyu_defense.py

```
@@ -0,0 +1,216 @@
+    def check_shiyu_countdown(self, screen: 'MatLike') -> bool:
+        """[复制自 shiyu_defense_battle.py:238] 防卫战倒计时 pipeline(4 contours=有倒计时)。"""
+        try:
+            # 检测普通倒计时
+            result1 = self.ctx.cv_service.run_pipeline('防卫战倒计时', screen, timeout=1.0)
+            has_countdown1 = result1 is not None and result1.is_success and len(result1.contours) == 4
+
+            # 检测精英倒计时
+            result2 = self.ctx.cv_service.run_pipeline('防卫战倒计时-精英', screen, timeout=1.0)
+            has_countdown2 = result2 is not None and result2.is_success and len(result2.contours) == 4
+
+            # 只要有一个倒计时被检测到，就认为有倒计时
```

Ruff (0.15.21)

[warning] Comment contains ambiguous `，` (FULLWIDTH COMMA). Did you mean `,` (COMMA)?

(RUF003)